contracts/multiswap-contracts/StargateComposer.sol

```diff
@@ -0,0 +1,122 @@
+
+abstract contract StargateComposer is Ownable, ILayerZeroComposer {
+    using OptionsBuilder for bytes;
+    using SafeERC20 for IERC20;
+    IStargate public stargate; // stargate usdc pool contract
+    IERC20 public usdc; // usdc contract address
+    address public endpoint; // endpoint contract of layer zero
+
+    event ReceivedMessage(uint256 amount, uint256 amountOut, bytes someMessage);
+
+    function initConfig(address _stargate, address _usdc, address _endpoint) external onlyOwner {
+        require(_stargate != address(0), "Stargate pool address cannot be zero");
+        require(_endpoint != address(0), "Stargate endpoint address cannot be zero");
+        require(_usdc != address(0), "USDC address cannot be zero");
+        stargate = IStargate(_stargate);
+        usdc = IERC20(_usdc);
+        endpoint = _endpoint;
+    }
+
+    function setEndpoint(address _endpoint) external onlyOwner {
+        require(_endpoint != address(0), "Stargate endpoint address cannot be zero");
+        endpoint = _endpoint;
+    }
+
+    function lzCompose(
+        address _from,
+        bytes32 _guid,
+        bytes calldata _message,
+        address _executor,
+        bytes calldata _extraData
+    ) external override payable {
+        require(_from == address(stargate), "!stargate");
+        require(msg.sender == endpoint, "!endpoint");
+
+        uint256 amountLD = OFTComposeMsgCodec.amountLD(_message);
+        bytes memory _composeMessage = OFTComposeMsgCodec.composeMsg(_message);
+
+        // Decode the _composeMessage to get the receiver address
+        address payee = abi.decode(_composeMessage, (address));
+
+        // Call the withdrawUSDC function on this contract
+        uint256 withdrawnAmount = withdrawUSDC(payee, amountLD);
+        emit ReceivedMessage(amountLD, withdrawnAmount, _composeMessage);
+    }
+
+    function withdrawUSDC(
+        address payee,
+        uint256 amount
+    ) private returns (uint256) {
+        usdc.safeTransfer(payee, amount);
+        return amount;
+    }
+
+    function swapUSDC(
+        uint32 _dstEid,
+        uint256 _amount,
+        address _composer,
+        bytes memory _composeMsg,
+        address _sourceAddress 
+    ) external payable {
+        uint256 valueToSend;
+        SendParam memory sendParam;
+        MessagingFee memory messagingFee;
+
+        (valueToSend, sendParam, messagingFee) = prepareTakeTaxi(_dstEid, _amount, _composer, _composeMsg);
+
+        require(msg.value >= valueToSend, "Insufficient gas sent with the transaction");
+
+        if(msg.value > valueToSend) {
+            uint256 gas = msg.value - valueToSend;
+            // Transfer the remaining gas fee to the user wallet
+            SafeAmount.safeTransferETH(_sourceAddress, gas);
+        }
+
+        usdc.approve(address(stargate), _amount);
+        stargate.sendToken{ value: valueToSend }(sendParam, messagingFee, _sourceAddress);
+    }
+    
+    function prepareTakeTaxi(
```

Comment:
Add some comments to this function. Hard to understand by just reading code. Stargate api is not very self descriptive

Reply:
Just added few comments as well as Natspecs.